fl4health/clients/basic_client.py

```diff
@@ -1319,10 +1319,12 @@ def load_checkpoint(self) -> None:
         # Optimizer is updated in setup_client to reference model weights from server
         # Thus, only optimizer state (per parameter values such as momentum)
         # should be loaded
-        for opt, state in zip(self.optimizers.values(), ckpt["optimizers_state"].values()):
-            opt_state_dict = opt.state_dict()
-            opt_state_dict["state"] = state
-            opt.load_state_dict(opt_state_dict)
+        for key in self.optimizers.keys():
+            optimizer = self.optimizers[key]
```

Comment:
super minor, but you can eliminate this line by doing `for key, optimizer in self.optimizers.items():`